### PR TITLE
0021: Create a KBI on publishing to Zenodo

### LINKED ISSUE #24: Is there a convienent workflow to publish a datalad dataset on zenodo?
Origin: DataLad chat, April 17th 2023

**Q:**  is there a convienent workflow to publish a datalad dataset on zenodo?

**A:**

CONP has a walkthrough in https://portal.conp.ca/share#datalad

Key steps, summarized
- publish the dataset to GitHub
- test whether a clone from Github can retrieve data
- login to Zenodo and add the Github repository
- create a [release](https://help.github.com/en/articles/creating-releases) on GitHub

DataLad handbook mentions Zenodo in [this footnote](http://handbook.datalad.org/en/latest/basics/101-139-hostingservices.html#id1)

TODO (not necessarily to be performed in this order)
- [x] Inform OP/Add reference to this issue at origin
- [x] Clarifying Qs asked or not needed
- [x] Nature of the issue is understood
- [x] Inform OP about resolution

### PR DESCRIPTION
**Pull request by mslw**
Both presented solutions are partial, but presenting them as they are is the best way to close #24; other than forking / re-implementing a special remote.

The original support issue was continued in https://github.com/datalad-handbook/book/issues/941 which serves both as a tracking issue for the broader problem and an overview of the current state of the existing (not ours) special remote implementation. The KBI links to that issue.

### CODE REVIEW

File: kbi/0021/index.rst [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+the dataset, no annexed files will be stored in Zenodo. Annexed file
+contents will need to be published to a special remote elsewhere.
+
+.. _`detailed instructions`: https://portal.conp.ca/share#datalad
+
+Partial solution 2: upload files using an existing special remote implementation
+--------------------------------------------------------------------------------
+
+We are aware of one git-annex special remote implementation for
+Zenodo, `git-annex-remote-zenodo`_. However, it is described as work
```
Comment (loj) [inline kbi/0021/index.rst:41]:
```suggestion
Zenodo, `git-annex-remote-zenodo`_. However, it is described as a work
```

File: kbi/0021/index.rst [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+in progress, and in our testing it showed limited functionality (with
+a small patch we were able to push files to Zenodo, and retrieve them
+back, but only in one and same dataset -- not in a dataset clone).
```
Comment (loj) [inline kbi/0021/index.rst:44]:
```suggestion
back, but only in one and the same dataset -- not in a dataset clone).
```

**Review by loj — APPROVED** [review]
This works for me. It's helpful to have something that references both solutions, even if they're partial.

I found a couple typos, but otherwise, I think it's good to go.

File: kbi/0021/index.rst [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+KBI0021: Publishing a DataLad dataset to Zenodo
+===============================================
+
+:authors: Michał Szczepanik <m.szczepanik@fz-juelich.de>
+:discussion: https://github.com/psychoinformatics-de/knowledge-base/pull/82
+:keywords: zenodo, special remote
+:software-versions: datalad_0.18.3
+
+This KBI discusses the currently available methods for publishing a
+dataset to `Zenodo`_.
```
Comment (jsheunis) [inline kbi/0021/index.rst:13]:
```suggestion
DataLad dataset to `Zenodo`_.
```

**Comment by jsheunis** [comment]
LGTM

**Review by jsheunis — APPROVED** [review]

**Pull request merged**